### ISSUE TITLE
NoMethodError: undefined method `gemspec' for nil:NilClass

### ISSUE THREAD

**Issue by pantocrator27** [mattbrictson/chandler#29]
I am trying to use chandler for a Chef Cookbook that I am authoring.  In including chandler's rake task, I am receiving the error message `NoMethodError: undefined method `gemspec' for nil:NilClass`.  I am not well versed in the ruby eco-system but I am thinking that this is because I do not have a gemspec file in the top level directory of my project.

**Comment by mattbrictson** [comment]
Yes, that's correct. The Chandler rake task assumes the presence of a gemspec.

I am not familiar with Chef, so I'm not sure what is expected here. Is a Chef Cookbook a Ruby gem that gets published to rubygems? If so, where is its version number specified? If the version is stored in a predictable location then integration with Chandler should be possible.

**Comment by pantocrator27** [comment]
I guess really the thing here is assuming a lot of people develop in Ruby and they are not necessarily developing Gems, then we may want to relax the requirement on the rake task to assume that there is a gemspec there and allow the developer, however they may see fit, to put the tag/version information in the task.

**Comment by mattbrictson** [comment]
Understood. However if you remove the gemspec auto-detect magic, then there isn't much left to the task at all. At that point you'd be better off writing a task from scratch, maybe? What part of the task would you like Chandler to provide out of the box? Could you provide some sample code (or pseudo-code) for how you'd like the Rakefile to look?

**Comment by pantocrator27** [comment]
Maybe the default behavior is the Raketask that does your gemspec auto-detect magic.  But then maybe the Raketask could be configured to accept a 'version' parameter that is up to the user to supply that would bypass the gemspec auto-detect magic ....

**Comment by mattbrictson** [comment]
Sorry, I still don't really understand how this would work. I'll leave this issue open in case someone is willing to take this one step further by proposing a PR.

**Comment by mattbrictson** [comment]
⚠️ **This project is no longer actively maintained.** I have found myself embracing tools like [release-drafter](https://github.com/apps/release-drafter) to keep release notes updated automatically, rather than relying on a manually edited CHANGELOG.md. If you are interested in taking over chandler development, let me know by [opening an issue](https://github.com/mattbrictson/chandler/issues/new).
